### ISSUE TITLE
Progress bar UI and accessibility issue

### ISSUE THREAD

**Issue by copiali** [reactstrap/reactstrap#1681]
- components: `progress`
- reactstrap version `#8.1.1`

Issue1.
### What is happening?

When value is greater than 100 (or less than 0), width will become greater than 100%(or negative).
https://github.com/reactstrap/reactstrap/blob/8.1.1/src/Progress.js#L68

### What should be happening?

Think we should set the max-width to be 100%. And fallback width to 0 if value is negative.

Issue2.
### What is happening?

No screenreader text for progress.

### What should be happening?

Should add screenreader text for progress.
I added `aria-valuetext={screenreaderText}` with aria-valuemin, aria-valuemax and aria-valuenow but it is not read aria-valuetext. 
After I removed aria-valuemin, aria-valuemax and aria-valuenow then aria-valuetext gets read out. Not sure why.

**Comment by TheSharpieOne** [comment]
for the first issue, it is really up to the user to provide logical and reasonable values. But I suppose it is fairly easy to constrain/bound/clamp the value to `0 <= percentage <= 100`.

For the second issue, we do not spread the rest of the props on the bar (we do it on the wrapper). I am not sure if other attributes would need to be added to the bar or not, but we should map `aria-valuetext` to the bar when present.

**Comment by quaj** [comment]
my fix is ```javascript style={{ width: `${Math.min(percent, 100)}%` }} ```
is this ok?

**Comment by jenniferharmon** [comment]
According to documentation:

> Authors SHOULD only set the aria-valuetext attribute when the rendered value cannot be accurately represented as a number. For example, a slider may have rendered values of small, medium, and large. In this case, the values of aria-valuenow could range from 1 through 3, which indicate the position of each value in the value space, but the aria-valuetext would be one of the strings: small, medium, or large.

If you want a more friendly or detailed read out, instead of using aria-valuetext you could consider having an aria-labelledby attribute that references a hidden (or visible) element on the page with more a detailed description. For example:

```
<div id="myLabel" class="hidden">My Current Download Progress:</div>
<div role="progressbar" aria-labelledby="myLabel" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100">
```

**Comment by copiali** [comment]
> According to documentation:

Yeah I think you are right. Maybe we can add aria-valuetext and screenreadertext as two optional props?

**Comment by TheSharpieOne** [comment]
> For the second issue, we do not spread the rest of the props on the bar (we do it on the wrapper). **I am not sure if other attributes would need to be added to the bar or not, but we should map aria-valuetext to the bar when present.**

**Comment by yogmel** [comment]
Hello everyone,

I see that this is labeled as a "good first issue", but I was wondering if I could work on this, as I have a task that needed the a11y feature.

Thanks!

**Comment by TheSharpieOne** [comment]
Based on PR #1787, I believe this has been resolved
